pkg/daemon/controller/controller.go

```diff
@@ -265,6 +276,86 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	return ctrl.Result{}, nil
 }
 
+// reconcileEnvironmentVariables makes sure that the proxy variables exist as expected, or are safely rectified.
+// Returns a boolean expressing whether the instance is awaiting a reboot.
+func (sc *ServiceController) reconcileEnvironmentVariables(envVars map[string]string, node core.Node) (bool, error) {
+	restartRequired, err := sc.ensureVarsAreUpToDate(envVars)
+	if err != nil {
+		return false, err
+	}
+	if restartRequired {
+		// Applying the reboot annotation results in an event picked up by WMCO's node controller to reboot the instance
+		if err = metadata.ApplyRebootAnnotation(sc.ctx, sc.client, node); err != nil {
+			return false, fmt.Errorf("error setting reboot annotation on node %s: %w", sc.nodeName, err)
+		}
+		return true, nil
+	}
+	// Wait until the environment variables at the process level are set as expected.
+	// This will only be picked up after WMCO reboots the instance
+	err = wait.PollUntilContextTimeout(sc.ctx, 15*time.Second, 5*time.Minute, true,
+		func(ctx context.Context) (done bool, err error) {
+			stillNeedsReboot := false
+			for _, varName := range cluster.SupportedProxyVars {
+				cmd := fmt.Sprintf("[Environment]::GetEnvironmentVariable('%s', 'Process')", varName)
+				out, err := sc.psCmdRunner.Run(cmd)
+				if err != nil {
+					return false, fmt.Errorf("error running PowerShell command %s with output %s: %w", cmd, out, err)
+				}
+				if strings.TrimSpace(out) != envVars[varName] {
+					stillNeedsReboot = true
+				}
+			}
```

Comment:
nit: this logic could be contained within its own function which would help readability of this larger function.

Reply:
Good point @sebsoto. Ideally there would be `waitForEnvironmentVariablesToReconcile()` function that calls the helper you are talking about.